src/utils.ts

```diff
@@ -0,0 +1,256 @@
+let JSONFallbackCache;
+
+export type Theme = {
+  [name: string]: {
+    [variable: string]: string;
+  };
+};
+
+/**
+ *
+ * @param {string} path
+ */
+export function loadCSS(path: string) {
+  const head = document.getElementsByTagName('head')[0];
+  const style = document.createElement('link');
+  style.href = path;
+  style.rel = 'stylesheet';
+  head.appendChild(style);
+}
+
+/**
+ *
+ * @param {string} style
+ */
+export function injectStyle(style: string) {
+  const styleTag = document.getElementById('themify') as HTMLLinkElement;
+  if (!styleTag) {
+    var node = document.createElement('style');
+    node.id = 'themify';
+    node.innerHTML = style;
+    document.head.appendChild(node);
+  } else {
+    styleTag.innerHTML = style;
+  }
+}
+
+/**
+ *
+ * .dark {
+ *   --primary-100: 30, 24, 33;
+ * }
+ *
+ * :root {
+ *   --primary-100: 22, 21, 22;
+ * }
+ *
+ * @param customTheme
+ * @returns {string}
+ */
+export function _generateNewVariables(customTheme: Theme) {
+  // First, we need the variations [dark, light]
+  const variations = Object.keys(customTheme);
+  return variations.reduce((finalOutput, variation) => {
+    // Next, we need the variation keys [primary-100, accent-100]
+    const variationKeys = Object.keys(customTheme[variation]);
+
+    const variationOutput = variationKeys.reduce((acc, variable) => {
+      const value = normalizeColor(customTheme[variation][variable]);
+      return (acc += `--${variable}: ${value};`);
+    }, '');
+
+    return (finalOutput += `${variation === 'light' ? ':root' : '.' + variation}{${variationOutput}}`);
+  }, '');
+}
+
+/**
+ *
+ * @returns {boolean}
+ */
+export function hasNativeCSSProperties() {
```

Comment:
I think it's better to catch the initial value and return it later, to save some time.